sdcm/cluster.py

```diff
@@ -4338,9 +4316,10 @@ def _rotate_kms_key(kms_key_alias_name, kms_key_rotation_interval, db_cluster):
                         message=f"Failed to rotate AWS KMS key for the '{kms_key_alias_name}' alias",
                         traceback=traceback.format_exc()).publish()
                 try:
-                    target_node = [node for node in db_cluster.nodes if not node.running_nemesis][0]
-                    self.log.debug("Target node for 'rotate_kms_key' is %s", target_node.name)
-                    with run_nemesis(node=target_node, nemesis_name="KMS encryption check"):
+                    nemesis_class = self.nemesis[0] if self.nemesis else getattr(
+                        import_module('sdcm.nemesis'), "Nemesis")
+                    with nemesis_class.run_nemesis(node_list=db_cluster.nodes, nemesis_label="KMS encryption check") as target_node:
```

Comment:
In parameters you provide list of nodes but in the context manager you save the result as `one` node.
Is it expected? If yes, why so? Looks not intuitive and confusing...

Reply:
it's expected, this context manager is selecting a node, so the logic of how to select a node that isn't during nemesis
is encapsulated in it.

not sure what the confusion here

Reply:
Strange logic, but o.k.